controllers/nicclusterpolicy_controller.go

```diff
@@ -176,23 +181,29 @@ func (r *NicClusterPolicyReconciler) updateNodeLabels(cr *mellanoxv1alpha1.NicCl
 			}
 		}
 	} else {
-		nodes := &corev1.NodeList{}
-		// We deploy OFED and Device plugins only on a nodes with Mellanox NICs
-		err := r.Client.List(context.TODO(), nodes, client.MatchingLabels{nodeinfo.NodeLabelMlnxNIC: "true"})
-		if err != nil {
-			return errors.Wrap(err, "failed to list nodes")
-		}
+		return r.clearMofedWaitLabel()
+	}
 
-		for i := range nodes.Items {
-			patch := []byte(fmt.Sprintf(`{"metadata":{"labels":{%q:"false"}}}`, nodeinfo.NodeLabelWaitOFED))
-			err := r.Client.Patch(context.TODO(), &nodes.Items[i], client.RawPatch(types.StrategicMergePatchType, patch))
-			if err != nil {
-				return errors.Wrapf(err, "unable to patch %s node label for node %s",
-					nodeinfo.NodeLabelWaitOFED, nodes.Items[i].Name)
-			}
-		}
+	return nil
+}
+
+func (r *NicClusterPolicyReconciler) clearMofedWaitLabel() error {
+	// We deploy OFED and Device plugins only on a nodes with Mellanox NICs
+	nodes := &corev1.NodeList{}
+
+	err := r.Client.List(context.TODO(), nodes, client.MatchingLabels{nodeinfo.NodeLabelMlnxNIC: "true"})
+	if err != nil {
+		return errors.Wrap(err, "failed to list nodes")
 	}
 
+	for i := range nodes.Items {
```

Comment:
can we only set it to false if label exist ?

that is, if a deployment doesnt use MOFED container, we will not add the label.

makes sense ?

Reply:
It will change the previous behavior that applied the label with `false` to all nodes with Mellanox NICs if Mofed is not set.
I am not sure what the impact will be.

Reply:
ack, lets keep as is